src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -115,6 +112,13 @@ object ApiOperations {
 
   }
 
+  def apply(dbHandle: Database): ApiOperations = new ApiOperations(dbHandle)
+
+}
+
+class ApiOperations(dbHandle: Database) {
```

Comment:
Instead of passing around a database handle all through the code, could we keep the old logic of fetching the handle when needed from `DatabaseUtil`? The code should be as stateless as possible. Every incoming query into Conseil provides the platform and network the query is for. Those can be provided to DatabaseUtil to get a handle each time. In order to prevent multiple IO operations 'DatabaseUtil` could cache the handles in a singleton instance.

Each public method of `ApiOperations` would have to take new `platform` and `network` parameters just as `TezosNodeOperator` does.

Reply:
Approach you are suggesting has issues which are really hard to overcome, service made this way may seem stateless but dependency on external service(DB in this case) that is hardcoded is not a good idea when it comes to testing. Also keeping DB in the one instance accessible by any other service makes reasoning about the architecture harder.

That's why I'm proposing to extract the database from the singleton object and pass it to the classes instead of putting platform and network to each method. @ivanopagano what do you think?